packages/core/src/types.ts

```diff
@@ -1960,3 +2008,13 @@ export type StateValueFrom<TMachine extends AnyStateMachine> = Parameters<
 
 export type StateFromMachine<TMachine extends AnyStateMachine> =
   TMachine['initialState'];
+
+export type PersistedMachineState<TState extends AnyState> = TState & {
+  children: {
+    [K in keyof TState['children']]: {
+      state: any; // TODO: fix (should be state from actorref)
+      src?: InvokeSourceDefinition;
```

Comment:
what do we get from using `InvokeSourceDefinition` here instead of a simple `string`?

Reply:
I'm actually going to move this to a simple `string` soon

Reply:
This is done in #3815